### DISCUSSION TITLE
Cache Corrupts Codes

### DISCUSSION THREAD

**Discussion by cemalgnlts** [vite-pwa/vite-plugin-pwa#632] [Q&A]
Hi,

The cache completes when the page is working properly, I install PWA and then refresh the page:

* First I see the message `onRegisteredSW` in the console.
* I see a few network requests.
* I see the `onRegisteredSW` message again.

As a result, the libraries I am using start giving errors.

I'm not sure if the problem is that I misused this library and `onRegisteredSW` works twice. But the problem only occurs when I give the files via sw and cache.

[main.ts](https://github.com/cemalgnlts/jside/blob/main/src/main.ts)
[vite.config.ts](https://github.com/cemalgnlts/jside/blob/main/vite.config.ts)

Demo: https://jside.vercel.app/

> [!Warning]
> Approximately **~65MB** of files will be cached when you use the demo.

**Comment by userquin** [comment]
![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/2ff5f79b-6106-4d66-8ca9-069f61823f34)

**Reply by cemalgnlts** [reply]
Did you install PWA and refresh the page?

**Reply by userquin** [reply]
It seems the main.ts being loaded twice, add a break point on `const Tee = Dee({`:

![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/4e7e8e29-0110-43fc-bb21-8672ee076cc5)

**Reply by userquin** [reply]
there is no need to install the PWA

**Reply by userquin** [reply]
why are you using the App this way? Why not just registering the App and call registerSW, you don't need to defer App load

**Reply by cemalgnlts** [reply]
That's how I first did it:
```diff
const updateSW = registerSW({
    immediate: true,
    onNeedRefresh() {
        console.log("onNeedRefresh()");
        if (confirm("New version available. Update?")) {
            updateSW(true);
        }
    },
    onRegisteredSW() {
        console.log("onRegisteredSW");
-       App();
    },
    onOfflineReady() {
        console.log("onOfflineReady");
    }
});

+ App();
```

When I started getting this error, I tried it that way in case I was using it wrong.

**Reply by cemalgnlts** [reply]
Yeah, everything seems to be loading twice, I'll take a look.

**Reply by userquin** [reply]
this is a `sideEffect`, no idea how you can solve it in your codebase

**Reply by userquin** [reply]
I'll take a look later (if I have time)...

**Reply by cemalgnlts** [reply]
Thanks.

**Reply by userquin** [reply]
There is no way, but it seems you cannot have any side effect in any module or importing any module with side effects, no idea if it is a react problem: I changed all src root modules to use dynamic import and still monaco loaded in the root dist module.

Now I'm getting this error on page refresh:
![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/bae1582e-9193-42dc-bb55-7b5fafbcbe3f)

**Reply by userquin** [reply]
I'll push my changes in a new branch in my fork (since you're using some env vars I switched the repo to pnpm to use `shell-emulator`)

**Reply by userquin** [reply]
Draft PR sent: https://github.com/cemalgnlts/jside/pull/3

**Reply by cemalgnlts** [reply]
Thank you. I will look into it (I don't use react by the way). I wonder what you mean by side effect.
